File: Camp Progress/215. Kth Largest Element in an Array.cpp
// Question Link: https://leetcode.com/problems/kth-largest-element-in-an-array/

// O(N + K)
class Solution {
public:
    int findKthLargest(vector<int>& nums, int k) {
        int counts[20001] = { 0 };
        for (int &num : nums) {
            counts[num + 10000]++;
        }
        
        vector<int> sorted;
        for (int i = 20000; i >= 0 ; i--) {
            while (counts[i]-- > 0) {
                sorted.push_back(i - 10000);
            }
        }
        
        return sorted[k - 1];
    }
};

// O(NlogN)
class Solution {
public:
    int findKthLargest(vector<int>& nums, int k) {
        sort(begin(nums), end(nums), greater<int>());
        return nums[k - 1];
    }
};

File: Python3 Solutions/215. Kth Largest Element in an Array.py
# Question Link: https://leetcode.com/problems/kth-largest-element-in-an-array/
class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        counts = [0]*20001
        for num in nums:
            counts[num + 10000] += 1
        
        kth = 0
        for index in reversed(range(20001)):
            kth += counts[index]
            if kth >= k:
                return index - 10000
                